### DISCUSSION TITLE
[Private Beta] GitHub Copilot code referencing

### DISCUSSION THREAD

**Discussion by 2percentsilk** [community/community#62595] [Copilot Conversations] [answered]
Here's your one-stop shop for FAQs around the code referencing private beta and the thread we'd love for you to leave feedback in!

### Tell me more about this feature!
Our [blog post](https://github.blog/2023-08-03-introducing-code-referencing-for-github-copilot/) breaks down this new private beta. 
TL;DR: When an accepted code suggestion has a match to code found in public repositories, we'll alert you and let you review the set of licenses associated with that repository as well as navigate to the repository itself.

### Where do I sign up?
Head to the [waitlist sign up page](https://github.co/copilot-code-referencing-waitlist)!

### It says I can't sign up for the waitlist. What's wrong?
Currently, this private beta is limited to our active Copilot for Individual users. 
If you're accessing Copilot via your organization with Copilot for Business, you are not eligible at this time. 
We hope to expand this preview to CfB users as soon as we can and thank you in advance for your patience!

### What editors are supported?
The experience is limited to Visual Studio Code for this stage of the beta but we're hard at work expanding the support to Visual Studio and IntelliJ.

### When can I expect to be admitted after signing up?
We can't share any specific timelines or guarantee access to this feature to all waitlist participants.

### Where can I get help with setup or questions once I'm in the beta?
You can submit feedback and solicit help by commenting in this discussion thread - please start a new comment for each item.

As this is a beta feature, please **_do not_** contact GitHub support as they cannot assist with troubleshooting this feature. They also cannot provide any timelines for admittance.

**Comment by SferaDev** [comment]
Some early feedback! Today while writing code I got the notification that public code matched, and actually it detected the same file in the remote repo (it's a public repository).

<img width="1104" alt="Screenshot 2023-08-10 at 08 53 41" src="https://github.com/community/community/assets/2181866/c4b3dc62-099d-4e23-8fa1-94817475b35c">

It would be nice if it ignored clones/forks of the repo you're working with.

**Reply by 2percentsilk** [reply]
Thanks for the feedback! We'll see how we can make this a bit less noisy 👍

**Reply by kralicky** [reply]
Every suggestion copilot makes seems to trigger this for the same repo I'm in. Until it's fixed, how do I disable this feature?

**Reply by 2percentsilk** [reply]
@kralicky sorry to hear that!
You can disable the feature preview by following these instructions: https://docs.github.com/en/get-started/using-github/exploring-early-access-releases-with-feature-preview. You can toggle this preview for yourself as needed in the Feature Previews dialog.

<img width="896" alt="image" src="https://github.com/community/community/assets/3174849/a6888dc6-0ac6-4e9b-bbf2-c2b56827ab02">

**Comment by sammcj** [comment]
I find this _really_ annoying, VSC already has a huge number of pop-ups and UI clutter which this just adds to.

- It shouldn't be a popup.
- You can't disable it in VSC.
- No instructions on why it's there or how to disable it.

**Reply by 2percentsilk** [reply]
Hey @sammcj ! How would you like to be notified about this?

Also you can disable this if you need! (https://github.com/orgs/community/discussions/62595#discussioncomment-6763719)

**Comment by alexdima** [comment]
I also find this really annoying! I am working on vscode, which is forked by very many people, and the code I write is flagged because of their forks. I would also like to suggest:
* avoid using popups and perhaps switch to something more subtle
* only compute such code referenced when accepting a Copilot suggestion that is non-trivial (more than 1 word)

**Reply by 2percentsilk** [reply]
We've got a redesign coming your way @alexdima - with a more integrated/subtle experience.

The acceptance on 1 word sounds like a bug....were you doing word by word acceptance with Copilot?

**Reply by alexdima** [reply]
Thanks for the quick reply! I wasn't doing word by word acceptance, I might have accepted something, but I can't recall accepting a meaningful block of code. It felt to me that the popup came without me doing anything, so I think perhaps I accepted a ghost text suggestion. But since I don't recall accepting something meaningful, I must have accepted one of those small(ish) suggestions like doing `console.log(` and it suggesting to log a local variable. I didn't really keep an exact track of what I did to trigger it.

**Comment by spaltrowitz** [comment, marked as answer]
This has now shipped in a public beta! https://github.blog/changelog/2023-11-30-github-copilot-november-30th-update/
